File: summer2019/week1/sort-array-by-parity.java
class Solution {
    public int[] sortArrayByParity(int[] A) {
        int[] res = new int[A.length];
        int lo = 0;
        int hi = res.length - 1;
        
        for (int i = 0; i < A.length; i++) {
            if (A[i] % 2 == 0) {
                res[lo] = A[i];
                lo++;
            } else {
                res[hi] = A[i];
                hi--;
            }
        }
        return res;
    }
}

File: summer2021/unit1/sort-array-by-parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        result = [0] * len(nums)
        lo = 0
        hi = len(nums) - 1
        
        for i in nums:
            if i % 2 == 0:
                result[lo] = i
                lo += 1
            else:
                result[hi] = i
                hi -= 1
        return result